src/configure/configure.ts

```diff
@@ -653,14 +643,100 @@ class Orchestrator {
             throw error;
         }
     }
+
+    private createProvisioningConfigurationObject(templateId: string, branch: string, pipeplineTemplateParameters: { [key: string]: string }, mode: provisioningMode): ProvisioningConfiguration {
+        return {
+          id: null,
+          branch,
+          pipelineTemplateId: templateId,
+          pipelineTemplateParameters: pipeplineTemplateParameters,
+          provisioningMode: mode,
+        } as ProvisioningConfiguration;
+      }
+
+    private async configurePipelineRemotely(): Promise<void>{
+        const provisioningConfigurer = new ProvisioningConfigurer(this.localGitRepoHelper);
+        const template = this.inputs.pipelineConfiguration.template as RemotePipelineTemplate;
+        try {
+            // prerequisite params
+            telemetryHelper.setCurrentStep('ConfiguringPreRequisiteParams');
+            await provisioningConfigurer.CreatePreRequisiteParams(this.inputs);
+
+            // Draft pipeline step
+            telemetryHelper.setCurrentStep('QueuedDraftProvisioningPipeline');
+            let provisioningServiceDraftModeResponse: ProvisioningConfiguration;
+            let draftProvisioningPipeline: ProvisioningConfiguration;
+            const provisioningConfiguration =  this.createProvisioningConfigurationObject(template.id, this.inputs.sourceRepository.branch, this.inputs.pipelineConfiguration.params , provisioningMode.draft );
+            await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: Messages.ConfiguringDraftPipeline }, async () =>
+                {
+                    try {
+                            // Initially send the provisioning request in draft mode to get workflow files
+                            provisioningServiceDraftModeResponse  = await provisioningConfigurer.queueProvisioningPipelineJob(provisioningConfiguration, this.inputs);
+                            if (provisioningServiceDraftModeResponse.id  != "" ) {
+                                // Monitor the provisioning pipeline
+                                telemetryHelper.setCurrentStep('AwaitDraftProvisioningPipeline');
+                                const OrgAndRepoDetails =  this.inputs.sourceRepository.repositoryId.split('/');
+                                draftProvisioningPipeline =  await provisioningConfigurer.awaitProvisioningPipelineJob(provisioningServiceDraftModeResponse.id, OrgAndRepoDetails[0], OrgAndRepoDetails[1], this.inputs );
+                            }else {
+                                throw new Error("Failed to configure provisioning pipeline");
```

Comment:
we should throw the error returned by the API

Reply:
We are throwing the same error returned by API. This else condition is for corner case where the return repsonose code is 2XX but we haven't received job id. In this case we are making an error

Reply:
got it.